x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoShape.java

```diff
@@ -40,4 +57,37 @@ public String toString() {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         return builder.value(shapeBuilder.toWKT());
     }
+
+    public static double distance(GeoShape shape1, GeoShape shape2) {
+        if (shape1.shapeBuilder instanceof PointBuilder == false) {
+            throw new SqlIllegalArgumentException("distance calculation is only supported for points; received [{}]", shape1);
+        }
+        if (shape2.shapeBuilder instanceof PointBuilder == false) {
+            throw new SqlIllegalArgumentException("distance calculation is only supported for points; received [{}]", shape1);
+        }
+        double srcLat = ((PointBuilder) shape1.shapeBuilder).latitude();
+        double srcLon = ((PointBuilder) shape1.shapeBuilder).longitude();
+        double dstLat = ((PointBuilder) shape2.shapeBuilder).latitude();
+        double dstLon = ((PointBuilder) shape2.shapeBuilder).longitude();
+        return GeoUtils.arcDistance(srcLat, srcLon, dstLat, dstLon);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
Can you use curly brackets for all these conditions, please? I think this is the overall style used in the other classes from SQL.

```
if (this == o) {
   return true;
}
if (o == null || getClass() != o.getClass()) {
   return false;
}
```